lms/models/user.py

```diff
@@ -47,3 +47,44 @@ class User(CreatedUpdatedMixin, Base):
 
     display_name: Mapped[str | None] = mapped_column(sa.Unicode, index=True)
     """The user's display name."""
+
+
+class LMSUser(CreatedUpdatedMixin, Base):
+    """A user record for each LMS user."""
+
+    __tablename__ = "lms_user"
+    __table_args__ = (sa.UniqueConstraint("tool_consumer_instance_guid", "lti_id"),)
+
+    id: Mapped[int] = mapped_column(autoincrement=True, primary_key=True)
+
+    tool_consumer_instance_guid: Mapped[str | None] = mapped_column(
+        sa.UnicodeText, index=True
+    )
+
+    h_userid: Mapped[str] = mapped_column(sa.Unicode, unique=True, index=True)
+    """The H userid which is created from LTI provided values."""
+
+    lti_id: Mapped[str] = mapped_column(sa.Unicode, index=True)
+    """ID of this user in the LMS, via LTI"""
```

Comment:
I think this is called the "user ID" in LTI, maybe `lti_userid` would be a better name for the column? Matches with `h_userid` and makes it clearer that this is a user ID and that there are two user IDs, one from `h_` and one from `lti_`